plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -236,7 +236,7 @@ public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
         verify(name.getTableType() == DATA, "Wrong table type: " + name.getTableType());
 
         Optional<Table> hiveTable = metastore.getTable(new HiveIdentity(session), tableName.getSchemaName(), name.getTableName());
-        if (hiveTable.isEmpty()) {
+        if (hiveTable.isEmpty() || getMaterializedView(session, tableName).isPresent() || getView(session, tableName).isPresent()) {
```

Comment:
`getMaterializedView || getView` could be cosultet earlier, before we get `Table` object.

now that we have the Table object, we could refer to it, instead of going by name again.

Also, the check could come before `verify(name.getTableType() == DATA` check

Please move the `getMaterializedView || getView`  and add maybe a comment that we rely on per-query caching, so this doesn't incur more communication to metastore.

Reply:
done